### PR TITLE
PMM-7888 TLS for PostgreSQL

### PR DESCRIPTION
**Pull request by percona-csalguero**
[PMM-7888 enabling percona to connect to Postgres using client certificates](https://jira.percona.com/browse/PMM-7888)

FB: https://github.com/Percona-Lab/pmm-submodules/pull/1898

- [pmm](https://github.com/percona/pmm/pull/749)
---
- [x] Tests passed.
- [ ] Feature build pass.
- [x] Update API dependency.

### CODE REVIEW

File: models/agent_model.go [hunk 1]
```diff
@@ -108,6 +108,19 @@ func (c AzureOptions) Value() (driver.Value, error) { return jsonValue(c) }
 // Scan implements database/sql.Scanner interface. Should be defined on the pointer.
 func (c *AzureOptions) Scan(src interface{}) error { return jsonScan(c, src) }
 
+// PostgreSQLOptions represents structure for special MySQL options.
+type PostgreSQLOptions struct {
+	TLSCa   string `json:"tls_ca"`
+	TLSCert string `json:"tls_cert"`
+	TLSKey  string `json:"tls_key"`
```
Comment (BupycHuk) [inline models/agent_model.go:115]: let's use ssl_ prefix since PostgreSQL uses that naming.

File: models/agent_model.go [hunk 2]
```diff
@@ -405,6 +419,20 @@ func (s *Agent) DSN(service *Service, dialTimeout time.Duration, database string
...
+			if s.PostgreSQLOptions != nil {
+				if files := s.Files(); len(files) > 0 {
+					for key := range files {
+						switch key {
+						case "tlsCa":
+							q.Set(key, tdp.Left+" .TextFiles.tlsCa "+tdp.Right)
+						case "tlsCert":
+							q.Set(key, tdp.Left+" .TextFiles.tlsCert "+tdp.Right)
+						case "tlsKey":
+							q.Set(key, tdp.Left+" .TextFiles.tlsKey "+tdp.Right)
```
Comment (BupycHuk) [inline models/agent_model.go:431]: as far as I see `s.Files()` returns
```
map[string]string{
				"ssl_ca_file":   s.PostgreSQLOptions.TLSCa,
				"ssl_cert_file": s.PostgreSQLOptions.TLSCert,
				"ssl_key_file":  s.PostgreSQLOptions.TLSKey,
			}
```
so this code will be skipped.
Please add more tests.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

**Review by JiriCtvrtka — CHANGES_REQUESTED** [review]
Here should be also pmm-agent PR for actions (like explain).

File: models/agent_model.go [hunk 3]
```diff
@@ -422,6 +437,22 @@ func (s *Agent) DSN(service *Service, dialTimeout time.Duration, database string
 		}
 		q.Set("sslmode", sslmode)
 
+		if s.PostgreSQLOptions != nil {
+			if files := s.Files(); len(files) > 0 {
+				for key := range files {
+					switch key {
+					case caFilePlaceholder:
+						//q.Set("ssl_ca_file", tdp.Left+".TextFiles.tlsCa"+tdp.Right)
```
[the remark this answers is not in the document]
Comment (askomorokhov) [reply models/agent_model.go:445]: Please removed commented line.

**Review by askomorokhov — APPROVED** [review]

**Review by JiriCtvrtka — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]

**Pull request merged**